### PR TITLE
Create MTHeaderService

### PR DESCRIPTION
**Pull request by menanteau**
Adding MTHeaderService for LSSTCam

### CODE REVIEW

File: bin/telemetry_sim_LSSTCam [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    parser.add_argument("--rotpa", action="store", default=0.0, type=float,
+                        help="The ROTPA for the visit")
+
+    parser.add_argument("--filterband", action="store", default='r',
+                        help="The filter band name (i.e. r)")
+    parser.add_argument("--filtername", action="store", default='U-001',
+                        help="Name of the physical filter (i.e. U-001)")
+    parser.add_argument("--filterslot", action="store", default=0,
+                        help="Filter slot (1-indexed)")
+    parser.add_argument("--filterpos", action="store", default=666.0,
```
Comment (mareuter) [inline bin/telemetry_sim_LSSTCam]: LSSTCam has a wheel not a slider, so not sure this is necessary here.

File: bin/telemetry_sim_LSSTCam [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    parser.add_argument("--ha", action="store", default='30.0', type=float,
+                        help="The HA")
+    parser.add_argument("--el", action="store", default=75.0, type=float,
+                        help="The elevation angle")
+    parser.add_argument("--az", action="store", default=45.0, type=float,
+                        help="The azimuth angle")
+    parser.add_argument("--focusz", action="store", default=13, type=float,
+                        help="Hexapod FocusZ position (mm)")
+
+    parser.add_argument("--instport", action="store", default=1, type=int,
```
Comment (mareuter) [inline bin/telemetry_sim_LSSTCam]: This and next entry are specific to LATISS and don't exist for LSSTCam.

File: etc/LSSTCam/primary_hdu.header [hunk 3]
```diff
@@ -0,0 +1,111 @@
+SIMPLE  =                    T /
+BITPIX  =                    8 / bits per data value
+NAXIS   =                    0 / number of axes
+EXTEND  =                    T / Extensions are permitted
+ORIGIN  = 'LSST DM Header Service'/ FITS file originator
+
+        ---- Date, night and basic image information ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header:5]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        ---- Image-identifying used to build OBS-ID ----
+CAMCODE =                      / The code for the camera (AT/CC/MC)
+CONTRLLR=                      / The controller (e.g. O for OCS, C for (A)CCS)
+DAYOBS  =                      / The observation day as defined by image name
+SEQNUM  =                      / The sequence number from the image name
+GROUPID =                      /
+TESTTYPE=                      / Test Type: BIAS:DARK:FLAT:OBS:PPUMP:QE:SFLAT
+EMUIMAGE=                      / Image being emulated
+
+        ---- Image sequence numbers
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+CURINDEX=                      / Index number for exposure within the sequence
+MAXINDEX=                      / Number of requested images in sequence
+
+        ---- Additional Keys Information from Camera ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+PROGRAM =                      / Name of the program
+REASON  =                      / Reason for observation
+
+        ---- Test Stand information ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+TSTAND  =                      / The Test Stand used
+
+        ---- Information from Camera (Common block) ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+IMAGETAG=                      / DAQ Image id
+OBSANNOT=                      / Observing annotation
+
+        ---- Filter/grating information ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+FILTBAND=                      / Name of the filter band
+FILTER  =                      / Name of the physical filter
+FILTSLOT=                      / Filter home slot
+FILTPOS =                      / Filter measured position of slide
+
+        ---- Exposure-related information ----
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: Section being provided by camera, remove.

File: etc/LSSTCam/primary_hdu.header [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+RADESYS =                      / Equatorial coordinate system FK5 or ICRS
+FOCUSZ  =                      / Focus Z position
+OBJECT  =                      / Name of the observed object
+
+        ---- Image-identifying used to build OBS-ID ----
+CAMCODE =                      / The code for the camera (AT/CC/MC)
+CONTRLLR=                      / The controller (e.g. O for OCS, C for (A)CCS)
+DAYOBS  =                      / The observation day as defined by image name
+SEQNUM  =                      / The sequence number from the image name
+GROUPID =                      /
```
Comment (mareuter) [inline etc/LSSTCam/primary_hdu.header]: This and next two entries provided by camera, remove.

File: etc/conf/mtTelemetry.yaml [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+imageParam_event:
+    device: MTCamera
+    topic: imageReadoutParameters
+    array_keys: ccdLocation
+
+# Telemetry values for images meta-data
+# https://confluence.lsstcorp.org/display/SYSENG/ComCam+Header+Information+Topic+Mapping
+telemetry:
+
+ DATE-OBS:
```
Comment (mareuter) [inline etc/conf/mtTelemetry.yaml]: This and next three entries are provided by the camera, do they need to be here?

File: etc/conf/mtTelemetry.yaml [hunk 12]
```diff
@@ -0,0 +1,484 @@
...
+  collect_after_event: start_collection_event
+
+ SEQNUM:
+  device: MTCamera
+  Stype: Event
+  topic: startIntegration
+  value: imageNumber
+  collect_after_event: start_collection_event
+
+ GROUPID:
```
Comment (mareuter) [inline etc/conf/mtTelemetry.yaml]: This entry and everything down to `--- Camera Sensor information` is provided by the camera. Do they need to be here?

**Review by mareuter — CHANGES_REQUESTED** [review]
Need to remove the camera provided information.

File: bin/telemetry_sim_LSSTCam (hunk 1 above)
Comment (menanteau) [reply bin/telemetry_sim_LSSTCam]: These have been removed

File: bin/telemetry_sim_LSSTCam (hunk 2 above)
Comment (menanteau) [reply bin/telemetry_sim_LSSTCam]: removed

File: etc/conf/mtTelemetry.yaml (hunk 11 above)
Comment (menanteau) [reply etc/conf/mtTelemetry.yaml]: Done -- but we need to make sure that Camera also computes `MJD-OBS`, `MJD-BEG` and `MJD-END`

File: etc/LSSTCam/primary_hdu.header (hunk 3 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header:5]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 9 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 4 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 5 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 6 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 7 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 8 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/LSSTCam/primary_hdu.header (hunk 10 above)
Comment (menanteau) [reply etc/LSSTCam/primary_hdu.header]: Removed

File: etc/conf/mtTelemetry.yaml (hunk 12 above)
Comment (menanteau) [reply etc/conf/mtTelemetry.yaml]: Removed

File: bin/telemetry_sim_LSSTCam [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+    parser.add_argument("--airmass", action="store", default=1.237, type=float,
+                        help="Airmass")
+    parser.add_argument("--ha", action="store", default='30.0', type=float,
+                        help="The HA")
+    parser.add_argument("--el", action="store", default=75.0, type=float,
+                        help="The elevation angle")
+    parser.add_argument("--az", action="store", default=45.0, type=float,
+                        help="The azimuth angle")
+
+    parser.add_argument("--m3port", action="store", default=1, type=int,
```
Comment (mareuter) [inline bin/telemetry_sim_LSSTCam]: Only necessary for AuxTel, can remove this.

File: etc/conf/mtTelemetry.yaml [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+    topic: endOfImageTelemetry
+
+## The event for the image readout paramters
+#imageParam_event:
+#    device: MTCamera
+#    topic: imageReadoutParameters
+#    array_keys: ccdLocation
+
+# Telemetry values for images meta-data
+# https://confluence.lsstcorp.org/display/SYSENG/ComCam+Header+Information+Topic+Mapping
```
Comment (mareuter) [inline etc/conf/mtTelemetry.yaml]: The link is pointing to the wrong Confluence page.

**Review by mareuter — APPROVED** [review]
Couple more minor things, but this looks like what's been requested.

File: bin/telemetry_sim_LSSTCam (hunk 13 above)
Comment (menanteau) [reply bin/telemetry_sim_LSSTCam]: Removed

File: etc/conf/mtTelemetry.yaml (hunk 14 above)
Comment (menanteau) [reply etc/conf/mtTelemetry.yaml]: Fixed

**Pull request merged**